src/hooks/useResizeObserverRef.ts

```diff
@@ -0,0 +1,49 @@
+import { useCallback, useEffect, useState } from "react";
+import type { CallbackRef, HTMLElementOrNull } from "../utils/utils";
+
+interface ResizeObserverOptions {
```

Comment:
Instead of using types from here, are there types from ResizeObserver natively somewhere? Just a thought.